src/components/PieChartComponent.tsx

```diff
@@ -0,0 +1,66 @@
+import Card from "@mui/material/Card/Card";
+import CardContent from "@mui/material/CardContent/CardContent";
+import Typography from "@mui/material/Typography";
+import { PieChart } from "@mui/x-charts/PieChart";
+import Switch from '@mui/material/Switch';
+import FormControlLabel from "@mui/material/FormControlLabel";
+import { useState } from "react";
+
+export type Data = {
+    id: number
+    value: number
+    label: string
+}
+
+export interface PieChartComponentProps{
+    data: any[]
+    width?: number  
+    height?: number
+    percentage?: boolean
+    title?: string
+}
+
+function convertDataToPercentage(data: Data[]): Data[] {
+    const totalValue = data.reduce((sum, entry) => sum + entry.value, 0);
+
+    const dataInPercentage = data.map(entry => ({
+        ...entry,
+        value: (entry.value / totalValue) * 100
+    }));
+
+    return dataInPercentage;
+}
+
+export default function PieChartComponent({data, width, height, title}: PieChartComponentProps) {
+    const [percentage, setPercentage] = useState(true);
+
+    return (
+    <Card sx={{ minWidth: 275 }}>
+        <CardContent>
```

Comment:
Rather than using the card element, things like a div or just `<></>` seem to be more appropriate here. The component would be chucked into the created dashboard card later, and a card wrapping a card does not seem that right (see screenshot)

![image](https://github.com/CamSIF/synergy-frontend/assets/95299427/891ed3b9-2e1f-45ed-b22b-db9e835ab17f)

Reply:
I tested removing it and it looks like this, which seems better
![image](https://github.com/CamSIF/synergy-frontend/assets/95299427/7fc2db12-f631-43a4-91eb-0a14f83d9165)